src/error.rs

```diff
@@ -81,6 +81,9 @@ pub enum ResolveError {
     /// Occurs when alias paths reference each other.
     #[error("Recursion in resolving")]
     Recursion,
+
+    #[error("{0} contains self-reference")]
```

Comment:
Tsc will only tell you " Cannot write file 'xxx/tsconfig.tsbuildinfo' because it will overwrite '.tsbuildinfo' file generated by referenced project", and you can not disable incremental behavior because of use the `references` config in tsconfig